src/PowerShellEditorServices/Language/FindDotSourcedVisitor.cs

```diff
@@ -32,15 +39,44 @@ public FindDotSourcedVisitor()
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitCommand(CommandAst commandAst)
         {
-            if (commandAst.InvocationOperator.Equals(TokenKind.Dot) &&
-                commandAst.CommandElements[0] is StringConstantExpressionAst)
+            CommandElementAst commandElementAst = commandAst.CommandElements[0];
+            if (commandAst.InvocationOperator.Equals(TokenKind.Dot))
             {
-                // Strip any quote characters off of the string
-                string fileName = commandAst.CommandElements[0].Extent.Text.Trim('\'', '"');
-                DotSourcedFiles.Add(fileName);
+                if (commandElementAst is StringConstantExpressionAst stringConstantExpressionAst)
```

Comment:
This could be a [pattern match switch](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching) if that is preferable.